src/main/java/soonwee/duke/Storage.java

```diff
@@ -0,0 +1,128 @@
+package soonwee.duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import java.util.Scanner;
+
+/**
+ * Represents a Storage instance. A storage instance will create new file,
+ * update file and read its data.
+ */
+public class Storage {
+
+    public String directory;
+    public File fileObject;
+    public TaskList taskList;
+
+    /**
+     * Instantiate Storage with its target directory.
+     */
+    public Storage(String directory) {
+        this.taskList = createNew();
+        this.directory = directory;
+        this.fileObject = createFile(directory);
+    }
+
+    /**
+     * Create new TaskList instance for this instance.
+     *
+     * @return a TaskList instance
+     */
+    public TaskList createNew() {
+        return new TaskList();
+    }
+
+    /**
+     * Create a file in the specified directory.
+     *
+     * @param directory target directory.
+     * @return new file object
+     */
+    public File createFile(String directory) {
+        try {
+            File fileObject = new File(directory);
+            if (fileObject.createNewFile()) {
+                System.out.println("New file created: " + fileObject.getName());
+            } else {
+                System.out.println("The file exists. Reading file...");
+                readFile();
+            }
+        } catch (IOException e) {
+            System.out.println("An error has occurred.");
+            e.printStackTrace();
+        }
+        return fileObject;
+    }
+
+    /**
+     * Read file contents.
+     */
```

Comment:
Perhaps you could leave this JavaDoc comment out, as it seems to be repeating the obvious.